### PR TITLE
[Quest]Tenshodo Membership Conv

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
- Moves Tenshodo Membership logic to IF. 
- Removes related NPC lua file logic.
- [Capture ](https://www.youtube.com/watch?v=jlvCW224OLU)shows quest is never accepted only completed.

## Steps to test these changes
- Can turn in either Tenshodo Invite to complete quest if unable to meet fame requirement, or accept quest and get the  application form.

### CODE REVIEW

File: scripts/quests/jeuno/Tenshodo_Membership.lua [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, { xi.item.TENSHODO_INVITE })
+                    then
```
Comment (Xaver-DaRed) [inline scripts/quests/jeuno/Tenshodo_Membership.lua]:
```lua
if npcUtil.tradeHasExactly(trade, { xi.item.TENSHODO_INVITE }) then
```

File: scripts/quests/jeuno/Tenshodo_Membership.lua [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+-- NOTE: Players can straight turn in Tenshodo Invite bypassing this quest. Quest never shows in log until completed.
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.JEUNO, xi.quest.id.jeuno.TENSHODO_MEMBERSHIP)
+
+quest.reward =
+{
+    item = xi.item.TENSHODO_INVITE,
+    keyItem = xi.ki.TENSHODO_MEMBERS_CARD,
+}
```
Comment (Xaver-DaRed) [inline scripts/quests/jeuno/Tenshodo_Membership.lua:16]: As I pointed out in a differen PR, please align this as so...
```lua
quest.reward =
{
    item    = xi.item.TENSHODO_INVITE,
    keyItem = xi.ki.TENSHODO_MEMBERS_CARD,
}
```

File: scripts/quests/jeuno/Tenshodo_Membership.lua [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            player:getFameLevel(xi.fameArea.JEUNO) >= 3
```
Comment (Xaver-DaRed) [inline scripts/quests/jeuno/Tenshodo_Membership.lua]:
```lua
        check = function(player, status, vars)
            return status == xi.questStatus.QUEST_AVAILABLE and
                player:getFameLevel(xi.fameArea.JEUNO) >= 3
```
This applies to all PRs aswell

File: scripts/zones/Lower_Jeuno/npcs/Ghebi_Damomohe.lua [hunk 4]
```diff
@@ -53,22 +26,6 @@ entity.onEventFinish = function(player, csid, option, npc)
         }
```
Comment (Xaver-DaRed) [inline scripts/zones/Lower_Jeuno/npcs/Ghebi_Damomohe.lua:27]: This is not related to your PR, but, while you are touching this file, could you please add a TODO comment for this items to be tabled to use xi.item enums?

No need to table them yourself unless you REALLY want to.

**Comment by Xaver-DaRed** [comment]
Looks good, just needs to correct some styling

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/quests/jeuno/Tenshodo_Membership.lua [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                player:getFameLevel(xi.fameArea.JEUNO) >= 3
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { xi.item.TENSHODO_INVITE }) then
+                        player:tradeComplete()
```
Comment (claywar) [inline scripts/quests/jeuno/Tenshodo_Membership.lua]: eating the items really should happen until the event finish

Comment (hooksta4) [reply scripts/quests/jeuno/Tenshodo_Membership.lua]: Good point. Moved.

**Pull request merged**